Document/0x07a-Testing-Authentication-and-Session-Management.md

```diff
@@ -305,7 +305,13 @@ Regular Expressions are often used to validate passwords. The password verificat
 
 Passwords can be set when registering accounts, changing the password or when resetting the password in a forgot password process. All of the available functions in the application that are able to change or set a password need to be identified in the source code. They should all be using the same password verification check, that is aligned with the password policy.
 
-If a frameworks is used that offers the possibility to create and enforce a password policy for all users of the application, the configuration should be checked.
+Here are different examples on how a validation can be implemented server-side:
+
+* Spring (Java) -  https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/validation/Validator.html
+* Ruby on Rails -  http://guides.rubyonrails.org/active_record_validations.html
+* PHP - http://php.net/manual/en/filter.filters.validate.php
```

Comment:
In PHP at least the normal validation functions are rarely used to enforce password requirements. I think these resources are too focused on general input validation and not enough on password requirement validation.

[Zxcvbn](https://github.com/dropbox/zxcvbn) is a checker specifically for passwords. Other than that, some [custom code](https://stackoverflow.com/questions/10752862/password-strength-check-in-php) is pretty common in PHP.